src/flatMap.ts

```diff
@@ -38,8 +38,16 @@ export function flatMap() {
   return purry(_flatMap, arguments, flatMap.lazy);
 }
 
+function _map<T, K>(array: Array<T>, mapFn: (value: T) => K): Array<K> {
+  const newArray = new Array<K>(array.length);
+  for (let i = 0; i < array.length; i++) {
+    newArray[i] = mapFn(array[i]);
+  }
+  return newArray;
+}
+
```

Comment:
Why reimplement map in the flatMap source and not use our own mapper (which uses the reduceLazy that was optimized). I feel like this complicates the code unnecessarily. At least I would add some documentation above this with a link to the article and some context on why it's better than mapper, otherwise a future maintainer might just refactor it back to a regular map, or even worse, if v8 ever becomes more efficient and using the build-in map would be better no one would clean this up